test-alerts.cpp

```diff
@@ -3,8 +3,45 @@
 #include "test/catch.hpp"
 #include "typewise-alert.h"
 
-TEST_CASE("infers the breach according to limits") {
-  double DegreesC = 20.0;
+TEST_CASE("Breach test for High active cooling, all temperatures") {
+  int DegreesC = HighActiveCoolingLimit;
   REQUIRE(inferBreach(CoolingType::HI_ACTIVE_COOLING, DegreesC)\
           == BreachType::NORMAL);
+  DegreesC = HighActiveCoolingLimit + 1;
+  REQUIRE(inferBreach(CoolingType::HI_ACTIVE_COOLING, DegreesC)\
+          == BreachType::TOO_HIGH);
+  DegreesC = LowLimit - 1;
+  REQUIRE(inferBreach(CoolingType::HI_ACTIVE_COOLING, DegreesC)\
+          == BreachType::TOO_LOW);
+}
+
+TEST_CASE("Breach test for Undefined cooling sollution") {
+  int DegreesC = HighActiveCoolingLimit;
+  REQUIRE(inferBreach(CoolingType::UNDEFINED_COOLING, DegreesC)\
+          == BreachType::UNDEFINED);
 }
+
+TEST_CASE("Handler test for all targets") {
+  BreachBaseHandler* handlerTester;
+  BreachHandlerFactory handlerFactory;
```

Comment:
`BreachHandlerFactory` hands out stubs for the email and controller. That makes it a part of the test functionality. So you could have the declarations in the `typewise-alert.h`, and the implementations of the stub in this test file itself.